include/qphix/blas_utils.h

```diff
@@ -42,6 +64,66 @@ cnmadd(FT res[2][S], const FT alpha[2], const FT x[2][S], const FT y[2][S])
   }
 }
 
+/**
+  Computes τ³ α x + β τ¹ x.
+
+  \param[out] res_up Array with complex index and then \c soalen index.
+  \param[out] res_dn Similar, down flavor
+  \param[in] alpha Complex number α
+  \param[in] beta Real number β
+  \param[in] x_up Input up flavor spinor
+  \param[in] x_dn Similar, down flavor
+
+  \tparam FT Floating point type
+  \tparam S soalen
+  */
+template <typename FT, int S>
+inline void
+tau3cm_tau1_scaleadd(FT res_up[2][S], FT res_dn[2][S], const FT alpha[2], const FT beta, 
+                     const FT x_up[2][S], const FT x_dn[2][S])
+{
+#pragma omp simd aligned(res_up, res_dn, x_up, x_dn : S)
+  for(int s = 0; s < S; s++) {
+    res_up[0][s] = alpha[0] * x_up[0][s] - alpha[1] * x_up[1][s];
+    res_up[1][s] = alpha[0] * x_up[1][s] + alpha[1] * x_up[0][s];
+
+    res_dn[0][s] = -alpha[0] * x_dn[0][s] + alpha[1] * x_dn[1][s];
+    res_dn[1][s] = -alpha[0] * x_dn[1][s] - alpha[1] * x_dn[0][s];
+    
+    // hopefully all x are still cached
```

Comment:
@bjoo: sorry for bothering you, but do you know if ```+=``` (no reduction here!) will work within ```#pragma omp simd```?